x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts

```diff
@@ -6,28 +6,35 @@
  */
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import createContainer from 'constate';
-import { useCallback, useReducer } from 'react';
+import { useCallback } from 'react';
 import { buildEsQuery, Filter, Query, TimeRange } from '@kbn/es-query';
 import DateMath from '@kbn/datemath';
 import type { SavedQuery } from '@kbn/data-plugin/public';
 import type { InfraClientStartDeps } from '../../../../types';
 import { useMetricsDataViewContext } from './use_data_view';
 import { useKibanaTimefilterTime } from '../../../../hooks/use_kibana_timefilter_time';
+import { useTimeRangeUrlState } from './use_time_range_url_state';
+import { useHostsQueryContext } from './use_host_query';
+import { useHostFiltersContext } from './use_host_filters';
 
 const DEFAULT_FROM_MINUTES_VALUE = 15;
 
 export const useUnifiedSearch = () => {
-  const [, forceUpdate] = useReducer((x: number) => x + 1, 0);
-
   const { metricsDataView } = useMetricsDataViewContext();
   const { services } = useKibana<InfraClientStartDeps>();
   const {
     data: { query: queryManager },
   } = services;
 
+  const { timeRange: selectedTimeRange, setTimeRange: setSelectedTimeRange } =
+    useTimeRangeUrlState();
+
+  const { applyFilterQuery } = useHostsQueryContext();
+  const { applyFilters } = useHostFiltersContext();
+
   const [getTime, setTime] = useKibanaTimefilterTime({
```

Comment:
We're calling this one 2x, here and in the `use_time_range_url_state.ts`. I think if you expose `from`, `to`, `getTime` and `setTime` in `use_time_range_url_state` and use them in this file, we might be able to remove this usage